pkg/endpoints/pipeline.go

```diff
@@ -112,44 +139,63 @@ func (r Resource) getPipeline(request *restful.Request, response *restful.Respon
 }
 
-/* Get all pipelines in a given namespace: the caller needs to handle any errors */
+/* Get all pipelines in a given namespace: the caller needs to handle any errors,
+an empty v1alpha1.Pipeline{} is returned if no pipeline is found */
 func (r Resource) getPipelineImpl(name, namespace string) (v1alpha1.Pipeline, error) {
 	logging.Log.Debugf("in getPipelineImpl, name %s, namespace %s", name, namespace)
 
 	pipelines := r.PipelineClient.TektonV1alpha1().Pipelines(namespace)
 	pipeline, err := pipelines.Get(name, metav1.GetOptions{})
 	if err != nil {
 		logging.Log.Errorf("could not retrieve the pipeline called %s in namespace %s", name, namespace)
-		return *pipeline, err
+		return v1alpha1.Pipeline{}, err
 	}
-	logging.Log.Debug("Found the pipeline definition OK")
+	logging.Log.Debugf("Found the pipeline definition OK")
 	return *pipeline, nil
 }
 
 /* Get all pipeline runs in a given namespace */
 func (r Resource) getAllPipelineRuns(request *restful.Request, response *restful.Response) {
 	namespace := request.PathParameter("namespace")
-	pipelineruns := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace)
 	repository := request.QueryParameter("repository")
 
 	logging.Log.Debugf("In getAllPipelineRuns: namespace: `%s`, repository query: `%s`", namespace, repository)
 
+	pipelinerunList, appError := r.GetAllPipelineRunsImpl(namespace, repository)
+	if appError.ERROR != nil {
+		logging.Log.Debugf("+%v", pipelinerunList.Items)
+		utils.RespondError(response, appError.ERROR, appError.CODE)
+	} else {
+		// no errors so return the pipeline run list
+		logging.Log.Debugf("+%v", pipelinerunList.Items)
+		response.WriteEntity(pipelinerunList)
+	}
+}
+
+// GetAllPipelineRunsImpl ...
+/* Returns a pointer to a list of PipelineRuns in a given namespace,
+optionally with a repository query as a parameter */
+func (r Resource) GetAllPipelineRunsImpl(namespace, repository string) (v1alpha1.PipelineRunList, AppError) {
+	logging.Log.Debugf("In getAllPipelineRunsImpl: namespace: `%s`, repository query: `%s`", namespace, repository)
+
+	pipelineruns := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace)
+
 	var pipelinerunList *v1alpha1.PipelineRunList
 	var err error
 	if repository != "" {
-		server, org, repo := getGitValues(repository)
+		server, org, repo, err := getGitValues(repository)
+		if err != nil {
+			logging.Log.Errorf("there was an error getting the Git values: %s", err)
+		}
 		match := gitServerLabel + "=" + server + "," + gitOrgLabel + "=" + org + "," + gitRepoLabel + "=" + repo
 		pipelinerunList, err = pipelineruns.List(metav1.ListOptions{LabelSelector: match})
-		logging.Log.Debugf("+%v", pipelinerunList.Items)
 	} else {
 		pipelinerunList, err = pipelineruns.List(metav1.ListOptions{})
-		logging.Log.Debugf("+%v", pipelinerunList.Items)
 		if err != nil {
-			utils.RespondError(response, err, http.StatusNotFound)
-			return
+			return v1alpha1.PipelineRunList{}, AppError{err, "", http.StatusNotFound}
 		}
```

Comment:
Please move this error check & return block outside of the else statement, because we check for an error getting the list here, but we do not check for an error getting the list with the `match` LabelSelector.

```suggestion
		}
	if err != nil {
		return v1alpha1.PipelineRunList{}, AppError{err, "", http.StatusNotFound}
	}
	return *pipelinerunList, AppError{}
```